src/cvodes/cvodea.c

```diff
@@ -2264,12 +2264,17 @@ static int CVAfindIndex(CVodeMem cv_mem, realtype t,
 
     *indx = ca_mem->ca_ilast;
     for(;;) {
+      if ( *indx >= ca_mem->ca_nsteps ) break;
       if ( sign*(t - dt_mem[*indx]->t) > ZERO) (*indx)++;
       else                                     break;
     }
 
     ca_mem->ca_ilast = *indx;
-
+    if ( *indx == ca_mem->ca_nsteps - 1 ) {
```

Comment:
This isn't quite right yet. What exactly should happen here at the right end of the array?